src/devdocs2zim/generator.py

```diff
@@ -491,6 +516,39 @@ def generate_zim(
             )
         return zim_path
 
+    @staticmethod
+    def fetch_logo_bytes(user_logo_path: str) -> bytes:
+        """Fetch a user-supplied logo for the ZIM and format/resize it.
+
+        Parameters:
+          user_logo_path: Path or URL to the logo.
+        """
+        logger.info(f"  Fetching logo from: {user_logo_path}")
+        full_logo_path: Path | None = handle_user_provided_file(source=user_logo_path)
+        if full_logo_path is None:
+            logger.warning("  Fetching logo failed, using fallback.")
```

Comment:
I don't think that fallbacking is a good thing from UX perspective. As a user, I would prefer the scraper to fail if the icon I asked to use is not fetchable for whatever reason. It would easily allow me to fix the URL if it has a typo typically. It feels misleading to produce a ZIM with a fallback icon.

Reply:
I was thinking that if you ran the scraper to fetch all current DevDocs resources you might see drift between that and icons you'd cached or that were in GitHub. I agree it may be confusing so I'll drop it.